objstore.go

```diff
@@ -756,7 +762,7 @@ func (r *timingReader) Read(b []byte) (n int, err error) {
 	r.readBytes += int64(n)
 	// Report metric just once.
 	if !r.alreadyGotErr && err != nil && err != io.EOF {
-		if !r.isFailureExpected(err) {
+		if !r.isFailureExpected(err) && r.ctx.Err() != context.Canceled {
```

Comment:
Have you considered using `errors.Is()` here?

Reply:
The other places that apply a similar special case check (eg. [here](https://github.com/thanos-io/objstore/blob/a1e58aa3bed5e446d4d66c8ce27973dc24d00850/objstore.go#L548) and [here](https://github.com/thanos-io/objstore/blob/a1e58aa3bed5e446d4d66c8ce27973dc24d00850/objstore.go#L533)) don't use `errors.Is()`, so I did the same thing here for consistency.

Reply:
I find it strange that the only reason for injecting the context is to check if it got cancelled, but the context is not used for the read operation. In practice a user can inject any context, even one which is not associated with the read.

Isn't the reader going to return `context.Cancelled` if the `Read` fails due to cancellation?

Reply:
Fair point - I've changed this in d44110f to check that the error returned satisfies `errors.Is(err, context.Canceled)`.